packages/web/app/src/lib/preflight/components/PreflightModal.tsx

```diff
@@ -0,0 +1,305 @@
+import { MonacoEditorReact } from '@/lib/MonacoEditorReact';
+import { Cross2Icon, InfoCircledIcon, TriangleRightIcon } from '@radix-ui/react-icons';
+import labApiDefinitionRaw from '../lab-api-declaration?raw';
+import { LogRecord, PreflightWorkerState } from '../shared-types';
+import { EditorTitle } from './EditorTitle';
+import { EnvironmentEditor } from './EnvironmentEditor';
+import { LogLine } from './LogLine';
+import { ScriptEditor } from './ScriptEditor';
+
+export function PreflightModal({
+  isOpen,
+  toggle,
+  content,
+  state,
+  execute,
+  abortExecution,
+  logs,
+  clearLogs,
+  onContentChange,
+  envValue,
+  onEnvValueChange,
+}: {
+  isOpen: boolean;
+  toggle: () => void;
+  content?: string;
+  state: PreflightWorkerState;
+  execute: (script: string) => void;
+  abortExecution: () => void;
+  logs: Array<LogRecord>;
+  clearLogs: () => void;
+  onContentChange: (value: string) => void;
+  envValue: string;
+  onEnvValueChange: (value: string) => void;
+}) {
+  const scriptEditorRef = useRef<editor.IStandaloneCodeEditor | null>(null);
+  const envEditorRef = useRef<editor.IStandaloneCodeEditor | null>(null);
+  const consoleRef = useRef<HTMLElement>(null);
+
+  const handleScriptEditorDidMount: MonacoEditorReact.OnMount = useCallback(editor => {
+    scriptEditorRef.current = editor;
+  }, []);
+
+  const handleEnvEditorDidMount: MonacoEditorReact.OnMount = useCallback(editor => {
+    envEditorRef.current = editor;
+  }, []);
+
+  const handleMonacoEditorBeforeMount = useCallback((monaco: MonacoEditorReact.Monaco) => {
+    // Configure JavaScript defaults for TypeScript validation
+    monaco.languages.typescript.javascriptDefaults.setDiagnosticsOptions({
+      noSemanticValidation: false,
+      noSyntaxValidation: false,
+      noSuggestionDiagnostics: false,
+      diagnosticCodesToIgnore: [], // Can specify codes to ignore
+    });
+    // monaco.languages.typescript.
+
+    // Enable modern JavaScript features and strict checks
+    monaco.languages.typescript.javascriptDefaults.setCompilerOptions({
+      allowNonTsExtensions: true,
+      allowJs: true,
+      checkJs: true,
+      // noEmit: true,
+      target: monaco.languages.typescript.ScriptTarget.ES2020,
+      // moduleResolution: monaco.languages.typescript.ModuleResolutionKind.NodeJs,
+      // module: monaco.languages.typescript.ModuleKind.ESNext,
+      lib: [],
+      // types: [],
+      strict: true,
+      noUnusedLocals: true,
+      noUnusedParameters: true,
+      noImplicitReturns: true,
+      noFallthroughCasesInSwitch: true,
+    });
+
+    // Add custom typings for globalThis
+    monaco.languages.typescript.javascriptDefaults.addExtraLib(
+      `
+        /// <reference lib="es2020" />
+
+        ${labApiDefinitionRaw}
+        declare const lab: LabAPI;
+
+
+        // ------------------------------------------------------------------------------------------------
+        // The following declarations are taken from:
```

Comment:
Let's move it to a file, just like the LabAPI, so we could format it, see what it defines easier, now it's just string that is super hard to read.

Reply:
Agreed upon solution of sing webworker lib removes need for this code in the first place now. 🎉